### PR TITLE
Refactoring of g_Config.m_SvTeam, See #4601

### PR DESCRIPTION
**Pull request by C0D3D3V**
This is just a small refactoring of g_Config.m_SvTeam as requested in #4601. Makes the code a little more readable. I'm not sure if `src/game/teamscore.h` is the best place for the enum (I'm not that familiar with the codebase yet), any opinions?

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/teamscore.h [hunk 1]
```diff
@@ -12,6 +12,14 @@ enum
 	VANILLA_TEAM_SUPER = VANILLA_MAX_CLIENTS
 };
 
+enum
+{
+	SRV_TEAM_FORBIDDEN,
```
Comment (def-) [inline src/game/teamscore.h:17]: Not sure I like the SRV_ prefix here. Maybe SV_TEAM_FORBIDDEN rather.

Comment (C0D3D3V) [reply src/game/teamscore.h:17]: SV_ prefix for all enum entries?

Comment (def-) [reply src/game/teamscore.h:17]: yes

File: src/game/teamscore.h [hunk 2]
```diff
@@ -12,6 +12,14 @@ enum
 	VANILLA_TEAM_SUPER = VANILLA_MAX_CLIENTS
 };
 
+enum
+{
+	SV_TEAM_FORBIDDEN,
```
Comment (def-) [inline src/game/teamscore.h:17]: might want to add the numbers here explicitly too, and add a comment not to change them.

**Comment by def-** [comment]
In the future you can add "(fixes #4601)" in the title, then it will get closed automatically on merge.

File: src/game/teamscore.h (hunk 2 above)
Comment (C0D3D3V) [reply src/game/teamscore.h:17]: I can do a PR for it tomorrow or the following days if no one else does it.

Comment (def-) [reply src/game/teamscore.h:17]: not that important, so I just merged it for now.

**Pull request merged**